prime-router/src/main/kotlin/azure/HistoryFunctions.kt

```diff
@@ -467,11 +468,17 @@ open class BaseHistoryFunction {
 
         if (jwtToken.isNotBlank()) {
             try {
-                val jwtClaims = JSONObject(String(Base64.getDecoder().decode(jwtToken.split('.')[1])))
-                userName = jwtClaims.getString("sub")
-                val orgs = jwtClaims.getJSONArray("organization")
+                // get the access token verifier
+                val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                    .setIssuer("https://${System.getenv("OKTA_baseUrl")}/oauth2/default")
+                    .build()
+                // get it to decode the token from the header
+                val jwt = jwtVerifier.decode(jwtToken)
+                // get the user name and org
+                userName = jwt.claims["sub"].toString()
+                val orgs = jwt.claims["organization"]
                 @Suppress( "UNCHECKED_CAST")
-                var org = if (orgs !== null) orgs.getString(0) else ""
+                val org = if (orgs !== null) (orgs as List<String>)[0] else ""
                 orgName = if (org.length > 3) org.substring(2) else ""
             } catch (ex: Throwable) {
                 System.out.println(ex)
```

Comment:
I would like the catch handler to log the exception instead of the print line. I would like it to also directly return null.